### ISSUE TITLE
Target Priorities are not standardized among unit types

### ISSUE THREAD

**Issue by apollodeathstar** [FAForever/fa#5919]
**Describe the bug**
Target Priorities are not standardized among unit types, possibly leading to either performance issues or less then optimal unit actions, effecting balance. 

**Expected behavior**
Units that share unit types have standardized targeting behavior at similar performance costs.

**Additional context**
Under guidance, I would attempt to standardize unit target priorities across unit types.

_This bug report will be read by people that have not experienced the bug themselves - this report is all they have to investigate. The more precise your report, the easier it becomes for us to help you._

**Comment by apollodeathstar** [comment]
I also think it may be good to add a new target priority category specifically for T1-3 interceptors/ASF then def Air units will be able to focus their defense on their biggest threats the ASF trying to kill them. 

The same category could be added to interceptors/ASF so that they focus other ASF's first but that may have to be checked for performance due to the very large numbers that get amassed in games.

I don't know if there has been much meta play regarding using T1 scouts etc in order to cheese/spoof enemy ASF balls but it is something to think about as a player behind on ASF numbers could tech pump out a bunch of scouts in order to make up for lack of ASF numbers in a fight to have enemy ASF waste salvos and win the fight. It is something to consider though. 

This may only be a issue in late game ASF vs ASF engagements so even just adding a category for ASF may be enough to ward off such an issue.

**Comment by apollodeathstar** [comment]
I agree that's the best way to go about it. splitting it into manageable categories will make it easier to submit and track and air units are a good place to start because it will be simpler then more complex interactions.

Is there a targetpriorities list somewhere that has the full list of possible targets? If not that is probably the first thing I can work on gathering from the various unit files.

**Comment by Garanas** [comment]
Target priorities are just (combined) categories. One could consider this to be a list:

- https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Core/Categories.lua

But in general for air units `GROUNDATTACK` means it is a gunship, `BOMBER` means it is a bomber, `TRANSPORTATION` means it is a transport and `ANTIAIR` means it shoots the previous three categories 😃

**Comment by Garanas** [comment]
> I also think it may be good to add a new target priority category specifically for T1-3 interceptors/ASF then def Air units will be able to focus their defense on their biggest threats the ASF trying to kill them.

This would be the `ANTIAIR` category

**Comment by apollodeathstar** [comment]
> > I also think it may be good to add a new target priority category specifically for T1-3 interceptors/ASF then def Air units will be able to focus their defense on their biggest threats the ASF trying to kill them.
> 
> This would be the `ANTIAIR` category

ANTIAIR covers all units capable of any sort of antiair fire from a tech 1 MAA to a broadsword with its secondary back mounted aa guns. I checked the wasp t3 uef asf and it did have in its categories ASF designation. I dunno if they all do so that's something I will have to check as well.

```
Categories = {
        "AIR",
        "ANTIAIR",
        "ASF",
        "BUILTBYTIER3FACTORY",
        "HIGHALTAIR",
        "MOBILE",
        "OVERLAYANTIAIR",
        "PRODUCTSC1",
        "RECLAIMABLE",
        "SELECTABLE",
        "TECH3",
        "UEF",
        "VISIBLETORECON",
    }
```

**Comment by Garanas** [comment]
> ANTIAIR covers all units capable of any sort of antiair fire from a tech 1 MAA to a broadsword with its secondary back mounted aa guns. I checked the wasp t3 uef asf and it did have in its categories ASF designation. I dunno if they all do so that's something I will have to check as well.

Units that are not designated as anti air but do have anti air weaponry should have the `WEAKANTIAIR` category, and not the `ANTIAIR` category.

Note that between layers (ground to air, air to air) is irrelevant because the weapon is limited by layer by default (that is unrelated to categories).

**Comment by apollodeathstar** [comment]
> > ANTIAIR covers all units capable of any sort of antiair fire from a tech 1 MAA to a broadsword with its secondary back mounted aa guns. I checked the wasp t3 uef asf and it did have in its categories ASF designation. I dunno if they all do so that's something I will have to check as well.
> 
> Units that are not designated as anti air but do have anti air weaponry should have the `WEAKANTIAIR` category, and not the `ANTIAIR` category.
> 
> Note that between layers (ground to air, air to air) is irrelevant because the weapon is limited by layer by default (that is unrelated to categories).

Makes sense when I am going through thru the units I will check the categories to make sure they are properly set for the unit.

**Comment by apollodeathstar** [comment]
> WEAKANTIAIR is added programmatically at the moment. It was only introduced recently to help AIs understand what units are worth building when you're looking for an anti air (or direct fire, etc) unit.
> 
> edit: what I'm trying to say is that it still involves some work to update the blueprints 😃 , I can tackle the UEF units this weekend.

That sounds good! then I will have your work to go off of and I can jump in and start helping after.

**Comment by apollodeathstar** [comment]
#5969 #5968 #5967 #5964 #5965

**Comment by Ejsstiil** [comment]
> WEAKANTIAIR is added programmatically at the moment. It was only introduced recently to help AIs understand what units are worth building when you're looking for an anti air (or direct fire, etc) unit.
> 
> edit: what I'm trying to say is that it still involves some work to update the blueprints 😃 , I can tackle the UEF units this weekend.

Hi @Garanas, this is the real world example how bad is any postprocessing done to the blueprints, as mentioned here: #5701
